services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v1/FlowController.java

```diff
@@ -358,4 +359,20 @@ public void invalidateFlowCache() {
         );
         return flowService.listFlowEvents(flowId, timeFrom, timeTo);
     }
+
+    /**
+     * Gets flow connected devices.
+     */
+    @ApiOperation(value = "Gets flow connected devices")
+    @GetMapping(path = "/{flow_id}/devices")
+    @ResponseStatus(HttpStatus.OK)
+    public CompletableFuture<FlowConnectedDevicesResponse> getHistory(
+            @PathVariable("flow_id") String flowId,
+            @ApiParam(value = "Device will be included in response if it's `time_last_seen` >= `since`. "
+                    + "Example of `since` value: `2019-09-30T16:14:12.538Z`",
+                    required = false)
+            @RequestParam(value = "since", required = false) Optional<String> since) {
+
+        return flowService.getFlowConnectedDevices(flowId, since.orElse(""));
```

Comment:
Why not parse the date-time field right in the controller?

Reply:
fixed